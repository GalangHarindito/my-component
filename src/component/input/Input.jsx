import React from "react";
import PropTypes from "prop-types";
import "./input.scss";

/**
 * Primary UI component for user interaction
 */

export const Input = ({
  label,
  name,
  size,
  placeholder,
  style,
  onChange,
  error,
  mandatory,
  className,
  ...props
}) => {
  return (
    <>
      <div className="wrapper-input">
        {label && (
          <label className="label-input" htmlFor="input">
            {label}
            {mandatory && <span className="asterix">*</span>}
          </label>
        )}
        <input
          type="text"
          id="input"
          placeholder={placeholder}
          name={name}
          className={[
            className,
            "storybook-input",
            `storybook-size-${size}`,
            `${error && `storybook-error-boundary`}`,
          ].join(" ")}
          style={style}
          onChange={onChange}
          {...props}
        />
        {error && (
          <div className="wrapper-error-message">
            <p className="error-input">{error}</p>
          </div>
        )}
      </div>
    </>
  );
};

Input.propTypes = {
  placeholder: PropTypes.string.isRequired,
  label: PropTypes.string,
  onChange: PropTypes.func,
  size: PropTypes.oneOf(["large", "medium"]),
  error: PropTypes.string,
  mandatory: PropTypes.bool,
};

Input.defaultProps = {
  onChange: undefined,
  placeholder: "Type Name",
  label: "Name",
  size: "medium",
  error: "",
  mandatory: false,
};
