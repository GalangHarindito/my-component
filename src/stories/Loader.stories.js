import { Loader } from "../component/spinner/loader";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
export default {
  title: 'Component/Loader',
  component: Loader,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes

  argTypes:{
    size: {
      description: 'Size for loader',
      action: 'string',
    }
  }
};

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args

export const Large = {
  args: {
    size: 'large',
  },
};

export const Medium = {
    args: {
      size: 'medium',
    },
  };

  export const Small = {
    args: {
      size: 'small',
    },
  };