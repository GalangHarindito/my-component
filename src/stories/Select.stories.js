import { SelectInput } from "../component/select/select";

export default {
    title: 'Component/Select',
    component: SelectInput,
    parameters: {
      // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
      layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
    tags: ['autodocs'],
    // More on argTypes: https://storybook.js.org/docs/api/argtypes
  
    argTypes:{
        onChange: {
            description: 'Handle behaviour when input event',
            action: 'type',
          },
          label: {
            description: 'Name for input label',
            action: 'string',
          },
          size: {
            description: 'Size for input',
            action: 'string',
          },
          error: {
            description: 'Error message if any',
            action: 'string',
          },
          required: {
            description: 'Required field',
            action: 'boolean',
          },
    }
  };
  
  // More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
  
  export const Select = {
    args: {
      size: 'large',
      label: 'Label Text',
      defaultValue: ' Input Value',
      error: '', 
      required: false,
      options: [
        { icon: "ðŸ¤£", value: "test1", name: "test1" },
        { icon: "ðŸ˜†", value: "test2", name: "test2" },
        { icon: "ðŸ¥°", value: "test3", name: "test3" },
      ]
    },
  };