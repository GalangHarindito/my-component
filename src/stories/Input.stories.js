import { Input } from "../component/input/Input";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
export default {
  title: 'Component/Input',
  component: Input,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes

  argTypes:{
    onChange: {
      description: 'Handle behaviour when input event',
      action: 'type',
    },
    placeholder: {
      description: 'Name for placeholder',
      action: 'string',
    },
    label: {
      description: 'Name for input label',
      action: 'string',
    },
    size: {
      description: 'Size for input',
      action: 'string',
    },
    error: {
      description: 'Error message if any',
      action: 'string',
    }
  }
};

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args

export const Large = {
  args: {
    size: 'large',
    label: 'Label Text',
    placeholder: 'Input Name',
    error: ''
  },
};

export const Medium = {
    args: {
      size: 'medium',
      label: 'Label Text',
      placeholder: 'Input Name',
      error: ''
    },
  };