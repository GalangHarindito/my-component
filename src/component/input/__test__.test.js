import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { composeStories } from '@storybook/react';
import * as stories from '../../stories/Input.stories';
import { Input } from './Input'; // import all stories from the stories file

const { Large, Medium } = composeStories(stories);

test('Make sure input UI does not change unexpectedly', () => {
    const { container } = render(<Large />)
    expect(container).toMatchSnapshot()
}) 

test('renders large input with default args', () => {
  render(<Large />);
  const inputPlaceholder = screen.getByPlaceholderText(
    /Input Name/i
  );
  const inputLabel = screen.getByLabelText(
    /Label Text/i
  )

  expect(inputPlaceholder).toHaveAttribute('placeholder', 'Input Name');
  expect(inputLabel).not.toBeNull();
});

test('renders input display value', () => {
  const onChange = jest.fn()

  render(<Input onChange={onChange} />);

  const inputValue = screen.getByDisplayValue('');
  fireEvent.change(inputValue, { target: { value: "Test Input" } });

  expect(onChange).toHaveBeenCalled()
});

test('Input should show validation', () => {
  render(<Large error='error' />);

  const errorDisplay = screen.findByText(
    "error"
  );

  waitFor(() => expect(errorDisplay).toBeInTheDocument());
});

test('Make sure input UI does not change unexpectedly', () => {
  const { container } = render(<Medium />)
  expect(container).toMatchSnapshot()
}) 

test('renders large input with default args', () => {
render(<Medium />);
const inputPlaceholder = screen.getByPlaceholderText(
  /Input Name/i
);
const inputLabel = screen.getByLabelText(
  /Label Text/i
)

expect(inputPlaceholder).toHaveAttribute('placeholder', 'Input Name');
expect(inputLabel).not.toBeNull();
});