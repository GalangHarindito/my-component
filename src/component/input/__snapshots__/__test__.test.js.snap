// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make sure input UI does not change unexpectedly 1`] = `
<div>
  <div
    class="wrapper-input"
  >
    <label
      class="label-input"
      for="input"
    >
      Label Text
    </label>
    <input
      class="storybook-input storybook-size-large "
      id="input"
      placeholder="Input Name"
      type="text"
    />
  </div>
</div>
`;

exports[`Make sure input UI does not change unexpectedly 2`] = `
<div>
  <div
    class="wrapper-input"
  >
    <label
      class="label-input"
      for="input"
    >
      Label Text
    </label>
    <input
      class="storybook-input storybook-size-medium "
      id="input"
      placeholder="Input Name"
      type="text"
    />
  </div>
</div>
`;
